icaruscode/CRT/CRTUtils/CRTT0MatchAlg.cc

```diff
@@ -793,5 +851,231 @@ namespace icarus{
+    }
+    return matchcanvec;
+    //auto tpcTrackHandle = event.getValidHandle<std::vector<recob::Track>>(fTPCTrackLabel);
+    //art::FindManyP<recob::Hit> findManyHits(tpcTrackHandle, event, fTPCTrackLabel);
+    //std::vector<art::Ptr<recob::Hit>> hits = findManyHits.at(tpcTrack.ID());
+    //return GetClosestCRTHit(detProp, tpcTrack, hits, crtHits);
+    //    for (const auto& match : matchedCan)
+    //return match;
+    //return nullmatch;
+  }
+*/
+
+    std::vector<icarus::match_geometry> CRTT0MatchAlg::GetClosestCRTHit_geo(detinfo::DetectorPropertiesData const& detProp,
+					    recob::Track tpcTrack, std::pair<double, double> t0MinMax, 
+					    std::vector<sbn::crt::CRTHit> crtHits, int driftDirection, uint64_t& trigger_timestamp, bool fIsData) {
+
+    auto start = tpcTrack.Vertex<TVector3>();
+    auto end   = tpcTrack.End<TVector3>();
+
+    float temp_start_x, temp_end_x; temp_start_x = start.X(); temp_end_x = end.X();
+    bool check_east_1 = temp_start_x < -210.29 && temp_end_x > -210.14;
+    bool check_east_2 = temp_end_x < -210.29 && temp_start_x > -210.14;
+    bool check_west_1 = temp_start_x < 210.14 && temp_end_x > 210.29;
+    bool check_west_2 = temp_end_x < 210.14 && temp_start_x > 210.29;
+	
+    bool check_east = check_east_1 || check_east_2;
+    bool check_west = check_west_1 || check_west_2;
+
+    bool simple_cathode_crosscheck = check_east || check_west;
+
+   int hit_id = 0;
+
+    // ====================== Matching Algorithm ========================== //
+    //  std::vector<std::pair<sbn::crt::CRTHit, double>> t0Candidates;
+    std::vector<match_geometry> t0Candidates;
+
+    //    if (crtHits.size() == 0) continue;
+    // Loop over all the CRT hits
+    for(auto &crtHit : crtHits){
+      // Check if hit is within the allowed t0 range
+      double crtTime = -99999.;  // units are us
+
+     icarus::match_geometry this_candidate = makeNULLmc_geo();
+     if(fIsData){
+      if (fTSMode == 1) {
+	crtTime = ((double)(int)crtHit.ts1_ns) * 1e-3; //+ fTimeCorrection;
+      }//end if (fTSMode == 1)
+      else {
+		crtTime = double(crtHit.ts0_ns - (trigger_timestamp%1000000000))/1e3;
+		if(crtTime<-0.5e6) 	crtTime+=1e6;
+		else if(crtTime>0.5e6) 	crtTime-=1e6;		
+      }//end else
+     }//end if(fIsData)
+     else if(!fIsData){	
+	crtTime = crtHit.ts0_ns/1e3;
+     }//end else if(!fIsData)
+      // If track is stitched then try all hits
+      if (!((crtTime >= t0MinMax.first - 10. && crtTime <= t0MinMax.second + 10.) 
+            || t0MinMax.first == t0MinMax.second)) continue;
+
+      // cut on CRT hit PE value
+      if (crtHit.peshit<fPEcut) continue;
+      if (crtHit.x_err>fMaxUncert) continue;
+      if (crtHit.y_err>fMaxUncert) continue;
+      if (crtHit.z_err>fMaxUncert) continue;
+      if (tpcTrack.Length() < fMinTrackLength) continue;
+
+      TVector3 crtPoint(crtHit.x_pos, crtHit.y_pos, crtHit.z_pos);
+
+      //Calculate Track direction
+      std::pair<TVector3, TVector3> startEndDir;
```

Comment:
Using `tracking::Point_t` as vector instead of `TVector3` _should_ avoid the need for some conversions.
Example later.